steghide: add page (#3950)
tldr-pages/tldr: pages/linux/steghide.md

# steghide

> Steganography tool for JPEG, BMP, WAV and AU file formats.
> More information: <https://github.com/StefanoDeVuono/steghide>.

- Embed data in a PNG image:

> `steghide embed --coverfile {{path/to/image.png}} --embedfile {{path/to/data.txt}} --passphrase "{{s3cr3t}}"`

Reviewer: I'd change `s3cr3t` to just `passphrase` or something else that's clear.

> `steghide embed --coverfile {{path/to/image.png}} --embedfile {{path/to/data.txt}} --passphrase "{{passphrase}}"`

Reviewer 2: Is there a way to prompt for the passphrase or have it stored in a file? This isn't particularly secure.

- Extract data from a WAV audio file:

`steghide extract --stegofile {{path/to/sound.wav}} --passphrase "{{passphrase}}"`

- Display file information, trying to detect an embedded file:

`steghide info {{path/to/file.jpg}}`

> - Embed data in a JPG image, compressing it at its maximum:

Reviewer 2: Perhaps this will read better?
```suggestion
- Embed data in a JPEG image, using maximum compression:
```

The `E` in `JPEG` is optional, but would make it more consistent with other pages (e.g. `jpegoptim`).

`steghide embed --coverfile {{path/to/image.jpg}} --embedfile {{path/to/data.txt}} --passphrase "{{s3cr3t}}" --compress {{9}}`

- Get the list of supported encryption algorithms and modes:

`steghide encinfo`

- Embed data in a JPG image, encrypted with Blowfish in ECB mode:

> `steghide embed --coverfile {{path/to/image.jpg}} --embedfile {{path/to/data.txt}} --passphrase "{{s3cr3t}}" --encryption {{blowfish}} {{ecb}}`

Reviewer: Are there other options similar to Blowfish and ECB? They can be listed as `{{blowfish|other}}` and `{{ecb|other}}` for example.

> Steganography tool for JPEG, BMP, WAV and AU file formats.
> More information: <https://github.com/StefanoDeVuono/steghide>.

> - Embed data in a PNG image, getting prompted for a passphrase:

Reviewer 2: Perhaps this will read better?
```suggestion
- Embed data in a PNG image, prompting for a passphrase:
```

`steghide encinfo`

> - Embed data in a JPEG image, encrypted with Blowfish in ECB mode:

Reviewer: I liked this one more. @sbrl What about you?

Author: @zdroid @sbrl And this way ?...

Reviewer 2: I think the new one is better. Note that you might need to go to the "Files changed" bit to see the latest version @zdroid, since GH's change viewing is being weird.

Reviewer: I did see it. It was added after this comment, but I felt no need to change my approval. :smile:
